build-scripts/compose_ds.py

```diff
@@ -29,6 +29,70 @@ def mangle_path(path):
     return path
 
 
+# When SCE support is enabled, retrieve the SCE checks and embed them in the datastream.
+# Collect every SCE check, extract the path to the script content, and include
+# every scripts in the datastream.
+def collect_sce_checks(datastreamtree, checklists, refdir):
+    ds_ns = '{' + datastream_namespace + '}'
+    xccdf_ns = '{' + XCCDF12_NS + '}'
+    sce_ns = '{' + sce_namespace + '}'
+    cat_ns = '{' + cat_namespace + '}'
+    xlink_href = '{' + xlink_namespace + '}' + 'href'
+
+    checklists_cref_location = '/'.join([
+        './',
+        ds_ns + 'checklists',
+        ds_ns + 'component-ref'
+    ])
+    checklists_component_ref = datastreamtree.find(checklists_cref_location)
+    # The component ID is the component-ref href without leading '#'
+    checklist_component_id = checklists_component_ref.get(xlink_href)[1:]
+
+    # Locate the <xccdf:check> inside <xccdf:Rule> and detect the SCE checks
+    checks_location = '/'.join([
+        './',
+        ds_ns + "component[@id='{}']".format(checklist_component_id),
+        xccdf_ns + 'Rule',
```

Comment:
I see a problem that the `Rule` element is never the direct child of the `component` element and that leads to situation that this XPath doesn't find anything and the rest of the fucntion would be void. `Rule` is usually a child of `Group` which can be inside another group and they are child of `Benchmark` which is the child of the source data stream component.